specification/indices/_types/IndexSettings.ts

```diff
@@ -227,7 +257,43 @@ export class IndexVersioning {
 }
 
 export class IndexSettingsLifecycle {
+  /**
+   * The name of the policy to use to manage the index. For information about how Elasticsearch applies policy changes, see Policy updates.
+   */
   name: Name
+  /**
+   * Indicates whether or not the index has been rolled over. Automatically set to true when ILM completes the rollover action.
+   * You can explicitly set it to skip rollover.
+   * @server_default false
+   */
+  indexing_complete?: boolean
+  /**
+   * If specified, this is the timestamp used to calculate the index age for its phase transitions. Use this setting
+   * if you create a new index that contains old data and want to use the original creation date to calculate the index
+   * age. Specified as a Unix epoch value in milliseconds.
+   * @server_default
+   */
+  origination_date?: long
+  /**
+   * Set to true to parse the origination date from the index name. This origination date is used to calculate the index age
+   * for its phase transitions. The index name must match the pattern ^.*-{date_format}-\\d+, where the date_format is
+   * yyyy.MM.dd and the trailing digits are optional. An index that was rolled over would normally match the full format,
+   * for example logs-2016.10.31-000002). If the index name doesn’t match the pattern, index creation fails.
+   */
+  parse_origination_date?: boolean
+  /**
+   * Time to wait for the cluster to resolve allocation issues during an ILM shrink action. Must be greater than 1h (1 hour).
+   * See Shard allocation for shrink.
+   * @server_default 12h
+   */
+  'step.wait_time_threshold'?: Time
```

Comment:
Are we sure that this field is **always** a "dotted" field? if it can also be represented as a nested field, then we should represent it as such.